### PR TITLE
Map coords takes a coordinate

### LINKED ISSUE #804: Replace MapCoords tuple parameter in function argument with a Coodinate parameter
https://github.com/georust/geo/pull/800#discussion_r849490674

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

FIXES #804 

Note there will be a little conflict with #836 - whoever merges second will have to make a small update.

### CODE REVIEW

File: geo/src/algorithm/map_coords.rs [hunk 1]
```diff
@@ -745,7 +726,10 @@ pub(crate) mod deprecated {
     use super::*;
 
     /// Map a fallible function over all the coordinates in a geometry, returning a Result
-    #[deprecated(since = "0.21.0", note = "use `MapCoords::try_map_coords` instead")]
+    #[deprecated(
+        since = "0.21.0",
+        note = "use `MapCoords::try_map_coords` which takes a `Coordinate` instead of an (x,y) tuple"
+    )]
     pub trait TryMapCoords<T, NT, E> {
```
Comment (michaelkirk) [inline geo/src/algorithm/map_coords.rs:733]: Note, for the deprecated methods, I left the signatures as `(T, T)` tuples, figuring they'll go away soon and this way we can better have the compiler tell the users what to do.

File: geo/src/algorithm/map_coords.rs [hunk 2]
```diff
@@ -41,28 +41,28 @@ mod modern {
...
         /// use approx::assert_relative_eq;
         ///
         /// let p1 = Point::new(10., 20.);
-        /// let p2 = p1.map_coords(|(x, y)| (x + 1000., y * 2.));
+        /// let p2 = p1.map_coords(|Coordinate { x, y }| Coordinate { x: x + 1000., y: y * 2. });
         ///
         /// assert_relative_eq!(p2, Point::new(1010., 40.), epsilon = 1e-6);
         /// ```
         ///
         /// You can convert the coordinate type this way as well
```
Comment (frewsxcv) [inline geo/src/algorithm/map_coords.rs:53]: Unrelated, but we should probs advertise `Convert` here now?

**Review by frewsxcv — APPROVED** [review]
lgtm, though you'll need to rebase/merge after my other pr merges

File: geo/src/algorithm/map_coords.rs (hunk 2 above)
Comment (michaelkirk) [reply geo/src/algorithm/map_coords.rs]: I've rebased, and pushed up a new commit which references `Convert` for the case of simply changing numeric type , and then updated this example to solve a more complex problem involving both type conversion and a transformation for which `Convert` would be insufficient.

**Comment by michaelkirk** [comment]
bors r=frewsxcv

**Pull request merged**